esrally/driver/runner.py

```diff
@@ -912,12 +955,38 @@ async def scroll_query(es, params):
                 "took": took
             }
 
-        if params.get("use-search-after"):
-            return await search_after_query(es, params)
-        elif "pages" in params:
-            return await scroll_query(es, params)
+        def _extract_search_after_properties(response: BytesIO, get_point_in_time: bool, hits_total: int) -> (dict, List):
+            properties = ["timed_out", "took"]
+            if get_point_in_time:
+                properties.append("pit_id")
+            # we only need to parse these the first time
+            if hits_total is None:
+                properties.extend(["hits.total", "hits.total.value", "hits.total.relation"])
+
+            parsed = parse(response, properties)
+
+            # standardize these before returning...
+            parsed["hits.total.value"] = parsed.pop("hits.total.value", parsed.pop("hits.total", 0))
+            parsed["hits.total.relation"] = parsed.get("hits.total.relation", "eq")
+
+            response_str = response.getvalue().decode("UTF-8")
+            index_of_last_sort = response_str.rfind('"sort"')
+            sort_pattern = r"sort\":([^\]]*])"
+            last_sort_str = re.search(sort_pattern, response_str[index_of_last_sort::])
+            if last_sort_str is not None:
+                last_sort = json.loads(last_sort_str.group(1))
+            else:
+                last_sort = None
+            return parsed, last_sort
+
+        search_method = params.get("operation-type")
+        if search_method == "paginated-search":
+            return await _search_after_query(es, params)
+        # also allow if pages is defined, for backward compatiblity
```

Comment:
I think we should deprecate the old approach after a grace period and log a warning here. Can you please raise an issue for this?

Reply:
deprecating now, opened #1215 for removal